uxarray/grid.py

```diff
@@ -277,8 +397,31 @@ def __from_ds__(self, dataset):
         else:
             raise RuntimeError("unknown mesh type")
 
+        # set precision flags
+        self._multi_precision = multi_precision
+        self._precision = precision
+
+        # loops through the dataset
+        if self._multi_precision:
+            for varName, varData in dataset.data_vars.items():
```

Comment:
Thanks for this Hongyu! I am still not sure why we wouldn't want to change everything in self.ds to multi-precision though. Are there some datasets inside there that can't be converted? Or do we just not use anything other than the node_y and node_x in calculations, so doing so would be a waste of computational power?

Reply:
1. We only want the multi-precision for floating point numbers. (Since our integer usage is not crazily big, we don't need the multi-precision integer so far)
2. the `self._ds` store many different data, I strongly recommend you take a closer look into this dataset
3. The helper function `convert_to_multi_precision()` function has a limited usage range, please read its docstring for more information